src/backend/src/services/teams.services.ts

```diff
@@ -162,4 +162,59 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  /**
+   * Update the given teamId's team's leads
+   * @param submitter a user who's making this request
+   * @param teamId a id of team to be updated
+   * @param userIds a array of user Ids that replaces team's old leads
+   * @returns an updated team
+   * @throws if the team is not found, the submitter has no privilege, or any user from the given userIds does not exist
+   */
+  static async setTeamLeads(submitter: User, teamId: string, userIds: number[]): Promise<Team> {
+    const team = await prisma.team.findUnique({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    const teams = await prisma.team.findMany(teamQueryArgs);
+
+    const newLeads = await getUsers(userIds);
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
+    if (!isAdmin(submitter.role) && submitter.userId !== team.headId) {
+      throw new AccessDeniedException('You must be an admin or the head to update the lead!');
+    }
+
+    if (newLeads.map((lead) => lead.userId).includes(team.headId)) {
+      throw new HttpException(400, 'A lead cannot be the head of the team!');
+    }
+
+    if (team.members.map((member) => member.userId).some((memberId) => userIds.includes(memberId))) {
+      throw new HttpException(400, 'A lead cannot be a member of the team!');
+    }
+
+    if (teams.map((team) => team.headId).some((teamHeadId) => newLeads.map((lead) => lead.userId).includes(teamHeadId))) {
+      throw new HttpException(400, 'A teams lead must not be a head of another team!');
```

Comment:
After a convo with Sofia and Walter, we do actually wanna allow this, so you can remove this check/test. Can you update the head endpoint as well to allow a head to also be a lead on other teams.